### PR TITLE
feat: Added Edit, Delete, and Trash page functionalities

### PR DESCRIPTION
**Pull request by yaswanthmyneni**
# Added Edit, Delete, and Trash page functionalities

### CODE REVIEW

File: src/components/note-card/NoteCard.jsx [hunk 1]
```diff
@@ -21,9 +26,22 @@ const NoteCard = ({ note }) => {
...
         <div className="flex gap-1rem">
           <MdLabel className="cursor label-icon" />
-          <MdOutlineEdit className="cursor edit-icon" />
+          <MdOutlineEdit
+            className="cursor edit-icon"
+            onClick={() => {
+              notesDispatch({ type: "DISPLAY_MODAL", payload: true });
+              notesDispatch({ type: "NOTE_ID", payload: _id });
+              notesDispatch({ type: "TITLE", payload: title });
+              notesDispatch({ type: "DESCRIPTION", payload: description });
```
Comment (Ankur9669) [inline src/components/note-card/NoteCard.jsx]: I am not sure why you have used 4 dispatch here??

File: src/context/trash-context.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const [trashState, trashDispatch] = useReducer(trashReducer, {
+    trashNotes: [],
+  });
+
+  const value = { trashState, trashDispatch };
+  return (
+    <TrashContext.Provider value={value}>{children}</TrashContext.Provider>
+  );
+};
+
```
Comment (Ankur9669) [inline src/context/trash-context.js:27]: Suggestion: You can create a custom hook useTrash here only

File: src/pages/notes-page/NotesPage.jsx [hunk 3]
```diff
@@ -30,8 +30,12 @@ const NotesPage = () => {
             <input type="text" className="input notes-input" />
```
Comment (Ankur9669) [inline src/pages/notes-page/NotesPage.jsx]: Use label for accesibilty

File: src/pages/notes-page/NotesPage.jsx [hunk 4]
```diff
@@ -30,8 +30,12 @@ const NotesPage = () => {
...
             <GoSettings className="filter-icon cursor" />
           </div>
-          {Object.keys(user).length > 0 &&
-            notes.map((note) => <NoteCard key={note._id} note={note} />)}
+          {notes?.length ? (
+            // Object.keys(user).length > 0 &&
```
Comment (Ankur9669) [inline src/pages/notes-page/NotesPage.jsx]: Remove commented code if not required

**Review by Ankur9669 — APPROVED** [review]
The functionalities are working fine and the code is also looking good, great work, i have given some small suggestions you can look them

**Review by omkarpatke — APPROVED** [review]
code is really good, readable do changes that Ankur gives, and feel free to merge.

**Pull request merged**